docs/doc/30-reference/20-functions/123-file-functions/presign.md

```diff
@@ -0,0 +1,28 @@
+---
+title: PRESIGN
+---
+
+Generates the pre-signed URL for a staged file by the stage name and file path you provide. The pre-signed URL enables you to access the file through a web browswer or an API request.
+
+:::tip
+This function currently works with files stored at Amazon S3 only.
+:::
+
+## Syntax
+
+```sql
+PRESIGN @<stage_name>/.../<file_name>
+```
+
+## Examples
+
+This example generates the pre-signed URL for the file `books.csv` on the stage `my-stage`:
```

Comment:
Please adds the following examples:

- `PRESIGN @my_stage/books.csv` get a download link
- `PRESIGN @my_stage/books.csv EXPIRE = 7200` get a download link which expired in 7200 seconds (2 hours)
- `PRESIGN DOWNLOAD @my_stage/books.csv` is the same with `PRESIGN @my_stage/books.csv`
- `PRESIGN UPLOAD @my_stage/books.csv` generate a link to upload as books.csv under `@my_stage`.